tests/conftest.py

```diff
@@ -0,0 +1,228 @@
+import threading
+import time
+import queue
+import sys
+import warnings
+import platform
+import itertools
+
+
+import pytest
+
+from PyQt5.QtWidgets import QApplication
+from PyQt5.QtCore import QTimer
+from _pytest.main import pytest_runtestloop as _pytest_runtestloop
+
+import ilastik.config
+
+from ilastik.utility.gui.threadRouter import ThreadRouter
+from ilastik.utility.itertools import pairwise
+from ilastik.shell.gui.startShellGui import launchShell
+
+
+GUI_TEST_TIMEOUT = 20  # Seconds
+
+
+def pytest_addoption(parser):
+    """Add command-line flags for pytest."""
+    parser.addoption("--run-legacy-gui", action="store_true",
+                     help="runs legacy gui tests")
+
+
+
+def pytest_pyfunc_call(pyfuncitem):
+    """
+    Defines protocol for legacy GUI test execution
+    """
+    if not is_gui_test(pyfuncitem):
+        return
+
+    bucket = [None]
+
+    def testfunc():
+        try:
+            return pyfuncitem.obj()
+        except Exception:
+            bucket[0] = sys.exc_info
+
+    thread = threading.Thread(target=testfunc)
```

Comment:
Is it possible to use futures and thread pools from [`concurrent.futures`](https://docs.python.org/3/library/concurrent.futures.html) here?

Reply:
Switched to concurrent futures for timeouts, thanks.